hat/assets/js/apps/Iaso/domains/stock/items/components/Filters.tsx

```diff
@@ -0,0 +1,277 @@
+import React, {
+    FunctionComponent,
+    useCallback,
+    useMemo,
+    useState,
+} from 'react';
+import SearchIcon from '@mui/icons-material/Search';
+import { Button, Grid } from '@mui/material';
+import { makeStyles } from '@mui/styles';
+import {
+    commonStyles,
+    useRedirectToReplace,
+    useSafeIntl,
+} from 'bluesquare-components';
+import DatesRange from 'Iaso/components/filters/DatesRange';
+import InputComponent from 'Iaso/components/forms/InputComponent';
+import { useDefaultSourceVersion } from 'Iaso/domains/dataSources/utils';
+import { OrgUnitTreeviewModal } from 'Iaso/domains/orgUnits/components/TreeView/OrgUnitTreeviewModal';
+import { useGetDataSources } from 'Iaso/domains/orgUnits/hooks/requests/useGetDataSources';
+import { useGetOrgUnit } from 'Iaso/domains/registry/hooks/useGetOrgUnit';
+import { useGetImpacts } from 'Iaso/domains/stock/hooks/useGetImpacts';
+import { useGetSkusDropdownOptions } from 'Iaso/domains/stock/hooks/useGetSkusDropdownOptions';
+import { Params } from 'Iaso/domains/stock/items/types/filters';
+import { formatParams } from 'Iaso/utils/requests';
+import MESSAGES from '../../messages';
+import { baseUrl } from '../config';
+
+const useStyles = makeStyles(theme => ({
+    ...commonStyles(theme),
+}));
+
+type Props = {
+    params: Params;
+};
+
+const ItemsFilters: FunctionComponent<Props> = ({ params }) => {
+    const [filtersUpdated, setFiltersUpdated] = useState(false);
+    const classes: Record<string, string> = useStyles();
+    const { formatMessage } = useSafeIntl();
+    const redirectToReplace = useRedirectToReplace();
+    const [filters, setFilters] = useState({
+        orgUnit: params.orgUnit,
+        sku: params.sku,
+    });
+
+    const { data: allSkus, isFetching: isFetchingSkus } =
+        useGetSkusDropdownOptions({});
+    const { data: dataSources } = useGetDataSources(true);
+    const sourceVersion = useDefaultSourceVersion();
+    const dataSource = useMemo(
+        () =>
+            dataSources?.find(
+                source => source.value === sourceVersion?.source?.id.toString(),
+            )?.value || '',
+        [dataSources, sourceVersion?.source?.id],
+    );
+    const { data: orgUnit } = useGetOrgUnit(filters.orgUnit?.toString());
+
+    const handleSearch = useCallback(() => {
+        if (filtersUpdated) {
+            setFiltersUpdated(false);
+            const tempParams: Params = {
+                ...params,
+                ...filters,
+            };
+            tempParams.page = '1';
+            redirectToReplace(baseUrl, formatParams(tempParams));
+        }
+    }, [filtersUpdated, params, filters, redirectToReplace]);
+
+    const handleChange = useCallback(
+        (key, value) => {
+            setFiltersUpdated(true);
+            setFilters({
+                ...filters,
+                [key]: value,
+            });
+        },
+        [filters],
+    );
```

Comment:
There's a `useFilterState` hook that provides all these state values and callbacks, ready to use with `InputComponent`. An example with the `OrgUnitTreeviewModal` can be found in `ReviewOrgUnitChangesFilter`